src/SilverStripe/TemplateParserBlockProvider.php

```diff
@@ -55,16 +57,18 @@ public static function cacheIncludeTemplate($res)
             throw new InvalidArgumentException('A template name must be passed into <% cache_include %>');
         }
 
-        $keyCreator = isset($res['Arguments'][1]) ? $res['Arguments'][1]['text'] : "'CacheIncludeKeyCreator'";
-        $cacheInclude = isset($res['Arguments'][2]) ? $res['Arguments'][2]['text'] : "'CacheInclude'";
+        $includeName = $res['Arguments'][0]['text'];
+        $configName = isset($res['Arguments'][1]) ? $res['Arguments'][1]['text'] : $includeName;
+        $keyCreator = isset($res['Arguments'][2]) ? $res['Arguments'][2]['text'] : "'CacheIncludeKeyCreator'";
+        $cacheInclude = isset($res['Arguments'][3]) ? $res['Arguments'][3]['text'] : "'CacheInclude'";
 
         return <<<PHP
-\$val .= \Injector::inst()->get($cacheInclude)->process(
-   {$res['Arguments'][0]['text']},
+\$val .= \SilverStripe\Core\Injector\Injector::inst()->get({$cacheInclude})->process(
+   {$configName},
```

Comment:
I need to update the readme for this, but the change looks more complicated than it is:

```
Old method (will still work)
<% cache_include 'Footer' %>

Causes issues with cache names (backslashes):
<% cache_include 'App\Includes\Footer' %>

Solution: provide cache block name separately:
<% cache_include 'App\Includes\Footer', 'Footer' %>
```